nemo/core/optim/optimizer_with_main_params.py

```diff
@@ -454,6 +454,10 @@ def step(self, **kwargs):
         if self._async_grad_allreduce:
             torch.cuda.synchronize()
 
+        closure = kwargs.pop('closure', None)
+        # Allows applications to specify closure kwarg without erroring due to duplicate specification
+        assert closure is None, f"closre should be None but was passed {closure}"
```

Comment:
```suggestion
        assert closure is None, f"closure should be None but was passed {closure}"
```

Reply:
the comment says this allows applications to specify the closure without erroring, but raises an assertion right after. n00b question: could you explain how the mcore optimizer closures are specified in this case?